apps/studio/components/layouts/EdgeFunctionsLayout/EdgeFunctionDetailsLayout.tsx

```diff
@@ -87,6 +102,26 @@ const EdgeFunctionDetailsLayout = ({
       ]
     : []
 
+  const downloadFunction = async () => {
+    if (filesError) return toast.error('Failed to retrieve edge function files')
+
+    const zipFileWriter = new BlobWriter('application/zip')
+    const zipWriter = new ZipWriter(zipFileWriter, { bufferedWrite: true })
+    functionFiles.forEach((file) => {
+      const fileName = file.name.split('/').pop() ?? file.name
```

Comment:
zip archives support directories right? I think if we include the full path they will be organized into directories.

ie. `const fileName = file.name`

Reply:
i'll try doing that actually

Reply:
i think ok to go ahead with current logic